xbmc/cores/VideoPlayer/VideoPlayerRadioID3.cpp

```diff
@@ -0,0 +1,359 @@
+/*
+ *  Copyright (C) 2005-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "VideoPlayerRadioID3.h"
+
+#include "Application.h"
+#include "DVDDemuxers/DVDDemuxUtils.h"
+#include "DVDStreamInfo.h"
+#include "GUIInfoManager.h"
+#include "ServiceBroker.h"
+#include "cores/VideoPlayer/Interface/TimingConstants.h"
+#include "guilib/GUIComponent.h"
+#include "music/tags/MusicInfoTag.h"
+#include "utils/EmbeddedArt.h"
+#include "utils/log.h"
+
+#include <taglib/attachedpictureframe.h>
+#include <taglib/commentsframe.h>
+#include <taglib/id3v1genres.h>
+#include <taglib/id3v2framefactory.h>
+#include <taglib/mpegfile.h>
+#include <taglib/tbytevectorstream.h>
+#include <taglib/textidentificationframe.h>
+
+using namespace TagLib;
+
+CDVDRadioID3Data::CDVDRadioID3Data(CProcessInfo& processInfo)
+  : CThread("DVDID3Data"),
+    IDVDStreamPlayer(processInfo),
+    m_speed(DVD_PLAYSPEED_NORMAL),
+    m_messageQueue("id3")
+{
+  CLog::Log(LOGDEBUG, "Radio ID3 tag processor - new {}", __FUNCTION__);
+
+  m_messageQueue.SetMaxDataSize(6 * 256 * 1024);
```

Comment:
make a note what those magic numbers mean or introduce constant(s)?

Reply:
I kinda wondered the same thing myself, I took this from audio player (RDS uses the size from VideoPlayer) assuming that it's reasonable to assume that one would never have more ID3 packets than audio packets.  Turns out the default value works just dandy, this line can seemingly just be removed (will test loading up the demux queue unreasonably to confirm)